_episodes/02-concepts.md

```diff
@@ -101,26 +145,40 @@ necessary computational environment for analysis.
 {: .callout}
 
 Alternatives that reproduce environments with minimal software 
-dependencies are technologies like virtual machines (virtualbox, vmwre, 
-NITRC CE), containers (docker, singularity) and installers (e.g., 
-vagrant, ansible). These can be very useful to replicate existing 
+dependencies are technologies like virtual machines (VirtualVox, VMWare, 
+NITRC CE), containers (Docker, Singularity) and installers (e.g., 
+Vagrant, Ansible). These can be very useful to replicate existing 
 environments and therefore simplify the installation problem 
 significantly. However, at present some of these technologies are not 
-installed by default on computing clusters you may have access to, 
-although that is changing.
+installed by default on computing clusters you may have access to.
 
 Lesson 3 covers container technologies and how to create, use, modify, 
 and reuse containers.
 
+> ## Exercise:
+>
+>  What should a reproducible computational environment contain?
+>
+> > ## Solution
+> >
+> > A reproducible computational environment must contain all the necessary data,
```

Comment:
Could we be more specific on "necessary data"? Will this include intermediate and output data besides input data?